### ISSUE TITLE
[Top Builder 2024]: BitScript

### ISSUE THREAD

**Issue by setzeus** [PlebLab/Top-Builder#23]
# 🏗️ Top Builder 2024 Application Form to track progress through Round 1 - 3 ~ Currently in Round 1

## 📝 Instructions
1. Only complete this form if you have been chosen for Top Builder, by PlebLab.
2. Provide detailed information to assist the judges in evaluating your project.

## 🧑‍🤝‍🧑 The Team
- **Team Name**: BitScript
- **Number of Founders**: 2
- **Team Background**:
We (setzeus & setbern) have together ran a development shop for ~10 years. Both of us are technical with setzeus focusing on product design & Bitcoin script items while setbern focuses on full-stack web development items. We also have an additional teammate who joined us recently who's focus is react/tailwind.
- **Bitcoin & Lightning Involvement**: Describe your involvement and experience.
We've been *around* Bitcoin with multiple consulting clients throughout the years, but this is the first time we're taking a proactive initiative with our own idea. 
- **Origin Story**: What inspired your idea?
We've recently worked in the Stacks ecosystem & slowly inched towards understanding Bitcoin script more deeply. Throughout this journey we realized all the missing Bitcoin development resources & tools. 

## 💡 The Idea
- **Project Overview**: What does your project do?
We train & empower Bitcoin developers by providing them with educational content & powerful development tools. Our focus is on specifically Bitcoin companies with multiple developers that are in constant need of access to Bitcoin development tools.
- **Website**: (Optional)
[https://www.bitscript.app/](https://www.bitscript.app/)
- **GitHub**:
Private (for now, will open-source our work as a library)
- **Project Stage**: Describe the current progress.
In alpha/pmf phase, we have a first cohort of customers that are actively helping us validate our future roadmap.
- **Supplementary Materials**: Provide any proof of work, workshops, or demo videos.
Best proof of work at the moment is the alpha linked above.

## 📚 Additional Information
- **Additional Links/Info**:

## 📌 Submission Type
- [x] Initial Project Creation

## 🚀 Current Project Progress
- **Progress Made**:
- **Challenges Faced**:
- **Future Roadmap of what's next**:

## ⏳ Round 1 Milestone
- **Keep in mind that completion will need to be by February 10th, 2024.**
- **First-round milestone the team has set for themselves:**
Have a clear next two quarter product roadmap for better serving teams (aka uncovering the biggest pain point).

## 📝 Submission Complete
- **All required information has been provided above.**

### Registration Opens: Monday, December 18, 2023
- **Application**: Interested teams apply for Top Builder.
- **Selection**: Teams will be chosen based on their application. See [FAQ](https://github.com/PlebLab/Top-Builder) for details.

### Round 1: January 13 - February 10, 2024
- **Workshops**: PlebLab weekly workshops start, available live and online.
- **Chat Access**: Selected teams join the PlebLab internal chat.
- **Mentorship**: Begins for teams.
- **Project Milestones**: Teams are set and judged on their first-round milestones.

### Round 2: February 10 - 24, 2024
- **Team Reduction**: PlebLab cuts half of the Top Builder teams. (Only half continued in Round 2.)
- **Project Milestones**: Teams are set and judged on their second-round milestones.
- **Workshops**: PlebLab weekly workshops continue.
- **Mentorship**: Continues.

### Round 3: Starting February 24, 2024
- **Team Reduction**: PlebLab narrows down to 6 Top Builder teams. (All will be pitching in Austin.)
- **Final Milestones**: Teams set final project milestones for the last round.
- **Preparation**: Teams prepare their pitch in front of judges. Final teams are contacted to come to Austin for Startup Day.

### Final Round at Startup Day: March 14, 2024
- **Preparation**: Teams work on their final pitches at PlebLab.
- **Presentations**: Final teams present before judges on Startup Day.
- **Deliberation**: Judges decide during a happy hour.
- **Announcement**: The winner of Top Builder is announced at Austin BitDevs.
- **Celebration**: After-party for Top Builder.

**Comment by setzeus** [comment]
**Week 1** (Jan 13th - 19th)
We kicked off this by designing three different possible user-flows / features that we wanted to narrow down for the remainder of the competition. We specifically worked on the following features:

[Taproot](https://www.figma.com/file/vDFNb7PuX37gJl7nzxFooU/BitScript-(latest)?type=design&node-id=3624-9168&mode=design)
We designed a wizard for manually creating a ScriptPath tree.

[Digital Signatures / PSBTs](https://www.figma.com/file/vDFNb7PuX37gJl7nzxFooU/BitScript-(latest)?type=design&node-id=6558-21835&mode=design)
We designed & explored the idea of creating a UI for all of the steps involved in a PSBT. This turned out to be deeply complex due to it's dependencies (we'd need to design a UI for signing & message digest).

[RPC](https://www.figma.com/file/vDFNb7PuX37gJl7nzxFooU/BitScript-(latest)?type=design&node-id=7512-17354&mode=design)
A Postman-like UI for learning & interacting with core RPC.

The goal of creating these three iterations is to prepare for the next week of user interviews.

**Comment by setzeus** [comment]
**Week 2** (Jan 20th - 26th)
During this week we primarily focused on user interviews. We interview our current first customer cohort along with a handful of new additional potential customers. 

The feedback was pretty universal & consistent: everything we've built so far is useful & is something users *want* to learn, but don't necessarily need. The reality we've learned is that most, if not all developers are actively working on the application layer. They do not need to read or write raw transactions in their-to-day.

However, whether it's directly or through an API-wrapper, *every* developer is interacting with RPC...

From this week we concluded to build out the RPC / Postman-like UI for interacting with core RPC!

**Comment by setzeus** [comment]
**Week 3** (Jan 27th - Feb 2nd)
This week contained two different tracks, one engineering & one marketing: 

The engineering track consisted of building out the MVP for the new RPC feature. We didn't wrap it up but should have something to demo by the following week.

The marketing track consisted of kicking off our [Twitter strategy](https://twitter.com/bitscriptapp). It led to our first ~400 followers & led to the first hints of a content/marketing machine to go from articles -> threads.

**Comment by setzeus** [comment]
**Week 4** (Feb 3rd - Feb 9th)
Two large updates on our end:

1. We had the chance to demo & conduct a user interview with one of our heroes. It was extremely enlightening as he pointed out the good, the bad, & the ugly.

2. [Our RPC MVP feature was completed!](https://bitscript-git-stage-setteam.vercel.app/rpc) As tracked above, this was the feature that we decided to build out next due to the feedback we gathered from our initial cohort of customers. It's currently in staging & we'll likely take a few more days of testing & polishing, but this will go to prod soon.

**Comment by setzeus** [comment]
**Week 4** (Feb 10th - Feb 16th)
One large update on our end:

We finally pushed the v0.1 of the RPC feature to production! You can find it [here](https://www.bitscript.app/rpc). While it's extremely simple at the moment, it's a big leap as it's the next large "developer" that we'll consistently update.

As mentioned above, some of the most consistent feedback from our first cohort of customers & prospect was always along the lines of "this is cool, I WANT to learn this but don't think we'd use it too often from day-to-day."

In short, for most the Bitcoin companies we'll be selling to, the majority of their developers will be working at the **API** level, not lower (which is where our current tools are the most powerful).

**Comment by setzeus** [comment]
### Round 2 Milestone(s)

While we slowed down our momentum a bit the last two weeks due to medical concerns, we're strapped in for a fiery next few week(s). We feel more comfortable with our product but feel like we need more work on the product, marketing & sales-side. Therefore...our next few goals are:

**1. Sharpening Our Content Machine** (marketing)
From the metrics we're tracking, our two big developer features (deserializer & sandbox) are by far the most heavily used. Right after, however, is the series of articles we've published that reside in the "Lessons" sections. It follows that the lessons/articles themselves are a powerful source of retention.

Additionally, it's become clear from this test Twitter thread, that articles -> thread is likely a powerful, consistent distribution strategy we can run; in short, every article is a form of content marketing that we previously underestimated. 

**Goal:** Define a process & plan out next 26 articles (w/ the idea of publishing one a week)

**2. Further Segmentation Exploration** (sales)
Throughout the product process we've identified & tried to serve multiple marketing segments. As of late, we've identified three different markets that we still need to conduct further research on:

1. Freelancers 
2. Educational Bitcoin Companies
3. General Bitcoin Companies

It's clear that focusing on 1. is a mistake since the TAM is small & can be served by focusing on 2. & 3. Deciding between 2. & 3. is particularly hard & require mores work. 

**Goal:** Reach out to 10+ BTC educational companies & 25+ general Bitcoin companies to gauge level of needs & possible next features.

**3. Next Iteration of RPC Feature** (product)
While we launched the v0.1, it's still extremely limited for everyday use. Curenttly, the main value prop is to use our RPC as a testing sandbox. This is useful occassionally, but not something that a team would use daily. 

**Goal:** Design out next version of RPC feature (v0.2).

**Comment by setzeus** [comment]
**Week 6** (Feb 17th - Feb 23rd)
Per our goals for the second round above, the last week focused much more on sales & marketing development than anything else (ie not as much engineering work). 

_Re: Lessons/Articles_
Our educational content powers both retention/time spent on site as well as our marketing strategy. Seen [here](https://x.com/bitscriptapp/status/1754134234107674640?s=46&t=55O7BQvrrwqgKJQPttWDAA), our first thread that was a direct downstream of an article we published led to ~12k views & 2 new customers. It's clear that we should streamline & automate this process for a weekly publishing cadence. In the next update I'll include the new articles we decided on.

_Re: Segmentation Exploration_
I spent the last two days hammering away on outbound DMs to relevant users that already follow our twitter account. We still haven't hit the goal above of 35+ entities; however, we are very happy to announce that some of these ideal entities have already contacted us back & we've had additional user interviews.

**Comment by setzeus** [comment]
### End of Round 2 Recap [draft]

Hi ya'll!

It's been an intense/exhilarating ~6 weeks & we're _extremely_ grateful for the experience. The community & energy have been nothing short of electric every step along the way & pushed us forward in many ways.

Below you'll find our checklist for the Round 2 Goals:

 ✅ After running a twitter poll & talking to a few community member we planned out the next 26-weeks / 6-months of articles/lessons
<img width="592" alt="Captura de pantalla 2024-02-22 a la(s) 10 09 27 p m" src="https://github.com/PlebLab/Top-Builder/assets/3859928/8b24c94b-5db7-4592-9451-c4b8c8132890">
<img width="493" alt="Captura de pantalla 2024-02-22 a la(s) 10 08 30 p m" src="https://github.com/PlebLab/Top-Builder/assets/3859928/3d45cb1e-72f1-4e99-9490-d1292c29226c">
